cookbook/views/api.py

```diff
@@ -661,6 +665,57 @@ def get_queryset(self):
         return queryset
 
 
+class AutoPlanViewSet(viewsets.ViewSet):
+    def create(self, request):
+        serializer = AutoMealPlanSerializer(data=request.data)
+
+        if serializer.is_valid():
+            keywords = serializer.validated_data['keywords']
+            start_date = serializer.validated_data['start_date']
+            end_date = serializer.validated_data['end_date']
+            meal_type = MealType.objects.get(pk=serializer.validated_data['meal_type_id'])
+            servings = serializer.validated_data['servings']
+
+            days = (end_date - start_date).days + 1
+            recipes = Recipe.objects.all()
+            meal_plans = list()
+
+            for keyword in keywords:
+                recipes = recipes.filter(keywords__name=keyword['name'])
+
+            recipes = recipes.order_by('?')[:days]
+            recipes = list(recipes)
+
+            for i in range(0, days):
+                day = start_date + datetime.timedelta(i)
+                recipe = random.choice(recipes)
+                args = {'recipe': recipe, 'servings': servings, 'title': recipe.name,
+                        'created_by': request.user,
+                        'meal_type': meal_type,
+                        'note': '', 'date': day, 'space': request.space}
+
+                m = MealPlan(**args)
+                meal_plans.append(m)
+
+            MealPlan.objects.bulk_create(meal_plans)
+            for m in meal_plans:
+                if request.data.get('addshopping', False) and request.data.get('recipe', None):
```

Comment:
i need to look at how the shopping stuff works but its very confusing so i will do it.